client/lib/feed-stream-store/index.js

```diff
@@ -45,6 +45,11 @@ function mixedKeyMaker( post ) {
 	return siteKeyMaker( post );
 }
 
+const recommendedKeyMaker = post => ( {
+	...siteKeyMaker( post ),
+	isRecommendation: true }
```

Comment:
@blowery and I talked about this more offline.
Final result is that for now, gonna move the `is_recommendation` away from the postKey and into a transform applied from the search-stream.